File: src/main/java/org/mitallast/queue/common/json/JsonModule.java
package org.mitallast.queue.common.json;

import com.google.inject.AbstractModule;
import org.mitallast.queue.common.codec.Codec;

public class JsonModule extends AbstractModule {
    static {
        Codec.register(6000, JsonMessage.class, JsonMessage.codec);
    }

    @Override
    protected void configure() {
        bind(JsonService.class).asEagerSingleton();
    }
}

File: src/main/java/org/mitallast/queue/common/json/JsonModule.kt
package org.mitallast.queue.common.json

import com.google.inject.AbstractModule
import org.mitallast.queue.common.codec.Codec

class JsonModule : AbstractModule() {

    override fun configure() {
        bind(JsonService::class.java).asEagerSingleton()
    }

    companion object {
        init {
            Codec.register(6000, JsonMessage::class.java, JsonMessage.codec)
        }
    }
}
